File: src/main/java/br/com/blz/testjava/api/controller/ProductController.java
package br.com.blz.testjava.api.controller;

import br.com.blz.testjava.api.domain.Product;
import br.com.blz.testjava.api.service.ProductService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/api/products")
public class ProductController {

    @Autowired
    private ProductService service;

    @PostMapping
    @ResponseStatus(HttpStatus.CREATED)
    public void createProduct(@RequestBody Product product) {
        service.save(product);
    }

    @GetMapping("/{sku}")
    public Product getProduct(@PathVariable("sku") Integer sku) {
        return service.findProductBySku(sku);
    }

    @PutMapping("/{sku}")
    @ResponseStatus(HttpStatus.NO_CONTENT)
    public void updateProduct(@RequestBody Product product, @PathVariable("sku") Integer sku) {
        service.update(product, sku);
    }

    @DeleteMapping("/{sku}")
    @ResponseStatus(HttpStatus.NO_CONTENT)
    public void deleteProduct(@PathVariable("sku") Integer sku) {
        service.delete(sku);
    }
}

File: src/main/java/br/com/blz/testjava/application/web/controller/ProductController.kt
package br.com.blz.testjava.application.web.controller

import br.com.blz.testjava.application.service.CreateProductService
import br.com.blz.testjava.application.service.DeleteProductService
import br.com.blz.testjava.application.service.GetProductService
import br.com.blz.testjava.application.service.UpdateProductService
import br.com.blz.testjava.application.web.controller.request.CreateProductRequest
import br.com.blz.testjava.application.web.controller.request.ProductResponse
import br.com.blz.testjava.application.web.controller.request.UpdateProductRequest
import org.springframework.web.bind.annotation.*

@RestController
class ProductController(
  private val createProductService: CreateProductService,
  private val getProductService: GetProductService,
  private val deleteProductService: DeleteProductService,
  private val updateProductService: UpdateProductService
) {
  @GetMapping("/product/{sku}")
  fun getProductBySku(
    @PathVariable("sku") sku: String
  ): ProductResponse {
    return getProductService.getProduct(sku)
  }

  @PostMapping("/product")
  fun createProduct(
    @RequestBody createProductRequest: CreateProductRequest
  ): ProductResponse {
    return createProductService.createProduct(createProductRequest)
  }

  @DeleteMapping("/product/{sku}")
  fun deleteProduct(
    @PathVariable("sku") sku: String
  ) {
    deleteProductService.deleteProduct(sku)
  }

  @PutMapping("/product/{sku}")
  fun deleteProduct(
    @PathVariable("sku") sku: String,
    @RequestBody updateProductRequest: UpdateProductRequest
  ) {
    updateProductService.updateProduct(sku, updateProductRequest)
  }
}
